### PR TITLE
Add params-extractor community module

### PR DESCRIPTION
**Pull request by nmco**

### CODE REVIEW

File: src/community/params-extractor/src/main/java/org/geoserver/params/extractor/Filter.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.logging.Logger;
+
+public final class Filter implements GeoServerFilter, ExtensionPriority {
+
+    private static final Logger LOGGER = Logging.getLogger(Filter.class);
+
+    private static List<Rule> rules;
+
+    public Filter(ResourceStore dataDirectory) {
+        Resource resource = dataDirectory.get("rules.xml");
```
Comment (mbarto) [inline src/community/params-extractor/src/main/java/org/geoserver/params/extractor/Filter.java]: Is the rules.xml stored in main data_dir folder? Can we move it to a more meaningful subfolder? (params-extractor) or give it a more meaningful name than rules.xml (params-extractor.xml)?

**Comment by mbarto** [comment]
Looks good to me, if we can just make the persisted file name / path more meaningful

**Comment by nmco** [comment]
Thank you for the review, you're right about the persistent path I will use the path 'params-extractor/extraction-rules.xml' if this sound good to you.

**Comment by mbarto** [comment]
yes, that path is good for me

**Comment by mbarto** [comment]
Adding some documentation in rst would be a nice addition

**Pull request merged**